mod/item.php

```diff
@@ -373,28 +310,19 @@ function item_post(App $a) {
 			}
 		}
 		if ($contact_id) {
-			$r = q("SELECT * FROM `contact` WHERE `id` = %d LIMIT 1",
-				intval($contact_id)
-			);
+			$author = dba::selectFirst('contact', [], ['id' => $contact_id]);
 		}
 	}
 
-	if (DBM::is_result($r)) {
-		$author = $r[0];
+	if (DBM::is_result($author)) {
 		$contact_id = $author['id'];
 	}
 
 	// get contact info for owner
-
 	if ($profile_uid == local_user() || $allow_comment) {
 		$contact_record = $author;
 	} else {
-		$r = q("SELECT * FROM `contact` WHERE `uid` = %d AND `self` LIMIT 1",
-			intval($profile_uid)
-		);
-		if (DBM::is_result($r)) {
-			$contact_record = $r[0];
-		}
+		$contact_record = dba::selectFirst('contact', [], ['uid' => $profile_uid, 'self' => true]);
```

Comment:
Isn't `DBM::is_result()` warranted here?

Reply:
In the previous coding ```$contact_record``` was undefined when the query had gone wrong. So here the variable is at least defined. It's not ideal but better than before.

Reply:
Gotcha.